Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -2507,8 +2507,13 @@ Bytecode::CodeGenerationErrorOr<Optional<Bytecode::Operand>> TryStatement::gener
     if (m_finalizer)
         generator.end_boundary(Bytecode::Generator::BlockBoundaryType::ReturnToFinally);
     if (m_handler) {
-        if (!m_finalizer)
-            unwind_context.emplace(generator, OptionalNone());
+        if (!m_finalizer) {
+            auto const* parent_unwind_context = generator.current_unwind_context();
```

Comment:
should we walk up the unwind contexts until there's one with a finalizer? or does grabbing the parent's finalizer handle this in a tail-cally way?

Also test262 diff plz